torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -847,7 +848,8 @@ ProcessGroupNCCL::CUDAEventCache& ProcessGroupNCCL::CUDAEventCache::get(
         it,
         std::piecewise_construct,
         std::forward_as_tuple(device),
-        std::forward_as_tuple());
+        std::forward_as_tuple(
+            std::make_shared<ProcessGroupNCCL::CUDAEventCache>()));
```

Comment:
Just a note: we may be able to simplify it in future now that we are not emplacing a mutex into the map. (Not required for this PR)

Reply:
oh right, we can use just emplace now. let me try.